pghoard/postgres_command.py

```diff
@@ -46,9 +54,19 @@ def restore_command(site, xlog, output, host=PGHOARD_HOST, port=PGHOARD_PORT):
         output_path = os.path.join(os.getcwd(), output)
         headers = {"x-pghoard-target-path": output_path}
         method = "GET"
-    conn = HTTPConnection(host=host, port=port)
-    conn.request(method, "/{}/archive/{}".format(site, xlog), headers=headers)
-    status = conn.getresponse().status
+    path = "/{}/archive/{}".format(site, xlog)
+
+    for retries in range(retry_count - 1, -1, -1):
+        try:
+            status = http_request(host, port, method, path, headers)
+            break
+        except socket.error as ex:
+            err = "HTTP connection to {0}:{1} failed: {2.__class__.__name__}: {2}".format(host, port, ex)
+            if not retries:
+                raise PGCError(err, exit_code=255)
```

Comment:
not really related to this changeset, but the restore command exit codes could be behind descriptive constant names